torch_xla/experimental/assume_pure.py

```diff
@@ -0,0 +1,121 @@
+import torch
+
+from inspect import signature
+from functools import wraps
+
+from torch_xla._internal.jax_workarounds import requires_jax
+import torch_xla.core.xla_builder as xb
+
+
+@requires_jax
+def assume_pure(fn):
+  """Decorates a pure PyTorch/XLA function to skip expensive re-tracing.
+
+  Returns a new function that will only be traced once for each unique
+  input tensor shapes or non-tensor input argument values. This is useful
+  for removing Lazy Tensor tracing overhead.
+  
+  The decorated function must be pure (i.e. no side-effects, behavior
+  only depends on inputs).
+
+  Limitations:
+  - The decorated function can only use upstream PyTorch operators e.g.
+    `torch.einsum`, `torch.nn.functional.layer_norm`. Custom PyTorch/XLA
+    operations such as `mark_sharding` are not supported. This limitation
+    may be lifted in the future.
+  """
+  from torchax.interop import jax_view
+  return j2t_autograd(jax_view(fn))
+
+
+@requires_jax
+def j2t_autograd(fn):
```

Comment:
actually I think this function makes sense to be inside of `torchax`.

This file can import it and call it there.

Torchax cannot import from this file because `@requires_jax` would disable the jax device?

Reply:
I'll move

Reply:
We could leave the `@requires_jax` in torch_xla land to keep a uni-directional dependency graph